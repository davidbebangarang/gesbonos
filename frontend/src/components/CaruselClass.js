import React  from 'react'
import SwiftSlider from 'react-swift-slider'

  class CaruselClass extends React.Component{

    img() {
      var data2 = this.props.imagen;
      //console.log(data2, "estas son las imagenes")
      var arr = [];
        for (var key in data2) {
        arr.push(data2[key]);
      }
      var data = [];
      for (var i =0;i<arr.length;i++){
        data.push({'id':`${i+1}`,'src':arr[i]})
      }
        return data;
    }
    componentDidMount() {  
      this.img();    
    }

    render() {  
      return(
        <div> 
        <div className="container">                                       
            <SwiftSlider data={this.img()}/>                 
        </div>
      </div>   
      );
    }

  }
  export default CaruselClass;