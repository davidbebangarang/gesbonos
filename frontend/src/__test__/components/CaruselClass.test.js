import React from 'react';
import { mount, shallow } from 'enzyme';
import { create }from 'react-test-renderer';
import ProviderMock from '../../__mocks__/ProviderMock';
import CaruselClass from '../../components/CaruselClass';

describe('<ViewCards />', () => {
  
  test('Render del componente CaruselClass', () => {
    const caruselClass = shallow(
      <ProviderMock>
        <CaruselClass />
      </ProviderMock>,
    );
    expect(caruselClass.length).toEqual(1);
  });
});
describe('Bono Snapshot', () => {
  test('CaruselClass la UI del componente Footer', () => {
    const caruselClass = create(
      <ProviderMock>
        <CaruselClass />
      </ProviderMock>,
    );
    expect(caruselClass.toJSON()).toMatchSnapshot();
  });
});