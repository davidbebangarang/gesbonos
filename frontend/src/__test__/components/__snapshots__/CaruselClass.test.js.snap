// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bono Snapshot CaruselClass la UI del componente Footer 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="sc-EHOje fEidzb"
    >
      <ul
        className="sc-bZQynM eYhPVL"
        height={450}
      />
      <ul
        className="sc-gzVnrw iSvKSU"
      />
      <div
        className="sc-bwzfXH sc-htpNat iLYhoA"
        height={450}
        onClick={[Function]}
      />
      <div
        className="sc-bwzfXH sc-bxivhb fzmTRd"
        height={450}
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
